graph-builder/src/graph.rs

```diff
@@ -90,9 +90,16 @@ pub fn run<'a>(opts: &'a config::Options, state: &State) -> ! {
 
     let mut runtime = tokio::runtime::current_thread::Runtime::new().unwrap();
 
+    // Don't wait on the first iteration
+    let mut period = std::time::Duration::from_secs(0);
+
     loop {
+        thread::sleep(period);
+        period = opts.period;
         debug!("graph update triggered");
 
+        let mut graph: Option<cincinnati::Graph> = None;
```

Comment:
I think you don't need this one, as `graph` is not persisted/accessed across loops here. A `let graph = match create_graph()` should be enough.

Reply:
I wanted to be explicit about this being possibly assigned by multiple statements in the loop. For now it's only one assignment but it'll be more soon. Anyway, I don't mind removing this and go for variable shadowing instead.

Reply:
If that is the case, then feel free to keep it. The comment above was because I'm generally skeptical on long-living mutable variables as it is easy to introduce logic bugs when the business logic gets more complex.

Reply:
It's arlready gone ;-) I'll just shadow it as many times as necessary